test/unit-test/CMakeLists.txt

```diff
@@ -1,92 +1,88 @@
-    endif()
-endif()
-
-# ============================ Test configurations =============================
-
-# Define a CMock resource path.
-set( CMOCK_DIR ${MODULE_ROOT_DIR}/test/unit-test/CMock CACHE INTERNAL "CMock library source directory." )
-
-# Include CMock build configuration.
-include( cmock_build.cmake )
-
-# Check if the CMock source directory exists, and if not present, clone the submodule
-# if BUILD_CLONE_SUBMODULES configuration is enabled.
-if( NOT EXISTS ${CMOCK_DIR}/src )
-    # Attempt to clone CMock.
-    if( ${BUILD_CLONE_SUBMODULES} )
-        clone_cmock()
-    else()
-        message( FATAL_ERROR "The required submodule CMock does not exist. Either clone it manually, or set BUILD_CLONE_SUBMODULES to 1 to automatically clone it during build." )
-    endif()
-endif()
-
-# Add unit test and coverage configuration.
-
-# Use CTest utility for managing test runs. This has to be added BEFORE
-# defining test targets with add_test()
-enable_testing()
-
-# Add build targets for CMock and Unit, required for unit testing.
-add_cmock_targets()
-
-# Add function to enable CMock based tests and coverage.
-include( ${MODULE_ROOT_DIR}/tools/cmock/create_test.cmake )
-
-# Include build configuration for unit tests.
-include( unit_test_build.cmake )
-
-# Add a target for running coverage on tests.
-add_custom_target( coverage
-    COMMAND ${CMAKE_COMMAND} -DCMOCK_DIR=${CMOCK_DIR}
-    -P ${MODULE_ROOT_DIR}/tools/cmock/coverage.cmake
-    DEPENDS cmock unity core_http_utest core_http_send_utest
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-)
+# ====================  Define your project name (edit) ========================
+set(project_name "core_http")
+
+# =====================  Create your mock here  (edit)  ========================
+
+# list the files to mock here
+list(APPEND mock_list
+            ${HTTP_INCLUDE_THIRD_PARTY_DIRS}/http_parser.h
+        )
+# list the directories your mocks need
+list(APPEND mock_include_list
+            ${HTTP_INCLUDE_THIRD_PARTY_DIRS}
+        )
+#list the definitions of your mocks to control what to be included
+list(APPEND mock_define_list
+            ""
+       )
+
+# ================= Create the library under test here (edit) ==================
+
+# list the files you would like to test here
+list(APPEND real_source_files
+            ${HTTP_SOURCES}
+        )
+# list the directories the module under test includes
+list(APPEND real_include_directories
+            .
+            ${HTTP_INCLUDE_PUBLIC_DIRS}
+            ${HTTP_INCLUDE_PRIVATE_DIRS}
+        )
+
+# =====================  Create UnitTest Code here (edit)  =====================
+
+# list the directories your test needs to include
+list(APPEND test_include_directories
+            .
+            ${HTTP_INCLUDE_PUBLIC_DIRS}
+            ${HTTP_INCLUDE_PRIVATE_DIRS}
```

Comment:
So if we remove `HTTP_INCLUDE_PRIVATE_DIRS`, the above will also need to be updated.

Reply:
This needs to be update in coreMQTT as well.